File: Assets/ProjectGame/Lib/data/class_data/CharacterImpression/BaseCharacterImpression.py
from ..BaseCustomClassData import BaseCustomClassData

class BaseCharacterImpression(BaseCustomClassData):
    def __init__(self):
        super().__init__()
        self.suspicion = 0.0  # 疑惑
        self.favorability = 0.0  # 友好度

    def read(self, reader):
        self.suspicion = reader.read_float()
        self.favorability = reader.read_float()

    def load_json(self, data):
        self.suspicion = data.get('suspicion', 0.0)
        self.favorability = data.get('favorability', 0.0)

File: Assets/ProjectGame/Lib/data/class_data/CharacterImpression/BaseCharacterImpression.js
import { BaseCustomClassData } from '../BaseCustomClassData.js';

export class BaseCharacterImpression extends BaseCustomClassData {
    constructor() {
        super();
        this.suspicion = 0.0; // 疑惑
        this.favorability = 0.0; // 友好度
    }

    read(view, offset) {
        let o = offset;
        this.suspicion = view.getFloat32(o, true); o += 4;
        this.favorability = view.getFloat32(o, true); o += 4;
        return o;
    }

    loadJson(data) {
        this.suspicion = data.suspicion ?? 0.0;
        this.favorability = data.favorability ?? 0.0;
    }
}
